tests/test_object_detection.py

```diff
@@ -0,0 +1,448 @@
+
+
+@pytest.mark.filterwarnings("ignore::UserWarning")  # Should be 2 warnings (first two calls)
+def test_prune_by_threshold():
+    pruned_predictions = _prune_by_threshold(predictions, 1.0)
+    print(pruned_predictions)
+    for image_pred in pruned_predictions:
+        for class_pred in image_pred:
+            assert class_pred.shape[0] == 0
+
+    pruned_predictions = _prune_by_threshold(predictions, 0.6)
+
+    num_boxes_not_pruned = 0
+    for image_pred in pruned_predictions:
+        for class_pred in image_pred:
+            if class_pred.shape[0] > 0:
+                num_boxes_not_pruned += 1
+    assert num_boxes_not_pruned == 46
+
+    pruned_predictions = _prune_by_threshold(predictions, 0.5)
+    for im0, im1 in zip(pruned_predictions, predictions):
+        for cl0, cl1 in zip(im0, im1):
+            assert (cl0 == cl1).all()
+
+
+def test_similarity_matrix():
+    ALPHA = 0.99
+    lab_bboxes, lab_labels = _separate_label(labels[0])
+    det_bboxes, det_labels, det_label_prob = _separate_prediction(predictions[0])
+
+    iou_matrix = _get_overlap_matrix(lab_bboxes, det_bboxes)
+    dist_matrix = 1 - _get_dist_matrix(lab_bboxes, det_bboxes)
+
+    similarity_matrix = iou_matrix * ALPHA + (1 - ALPHA) * (1 - dist_matrix)
+    assert (similarity_matrix.flatten() >= 0).all() and (similarity_matrix.flatten() <= 1).all()
+
+
+def test_compute_label_quality_scores():
+    scores = _compute_label_quality_scores(labels, predictions)
+    scores_with_threshold = _compute_label_quality_scores(labels, predictions, threshold=0.9)
+    assert np.sum(scores) != np.sum(scores_with_threshold)
+
+    min_pred_prob = _get_min_pred_prob(predictions)
+    scores_with_min_threshold = _compute_label_quality_scores(
+        labels, predictions, threshold=min_pred_prob
+    )
+    assert (scores == scores_with_min_threshold).all()
+
+
+def test_find_label_issues():
+    auxiliary_inputs = _get_valid_inputs_for_compute_scores(ALPHA, labels, predictions)
+
+    overlooked_scores_per_box = _compute_overlooked_box_scores(
+        alpha=ALPHA,
+        high_probability_threshold=HIGH_PROBABILITY_THRESHOLD,
+        auxiliary_inputs=auxiliary_inputs,
+    )
+
+    overlooked_scores_no_auxillary_inputs = _compute_overlooked_box_scores(
+        alpha=ALPHA,
+        high_probability_threshold=HIGH_PROBABILITY_THRESHOLD,
+        labels=labels,
+        predictions=predictions,
+    )
+    for score, no_auxiliary_inputs_score in zip(
+        overlooked_scores_per_box, overlooked_scores_no_auxillary_inputs
+    ):
+        assert (score == no_auxiliary_inputs_score).all()
+
+    overlooked_issues_per_box = _find_label_issues_per_box(
+        overlooked_scores_per_box, OVERLOOKED_THRESHOLD
+    )
+    overlooked_issues_per_image = _pool_box_scores_per_image(overlooked_issues_per_box)
+    overlooked_issues = np.sum(overlooked_issues_per_image)
+    assert np.sum(overlooked_issues_per_image[5:]) == 5  # check bad labels were detected correctly
```

Comment:
Can we actually check the individual indices are correct?

Reply:
This checks individual indices were correct since indices 5,6,7,8,9 (`[5:]`) were issues and the next line checks only 5 issues were detected.